cloud/deployment/deployment.go

```diff
@@ -171,6 +171,16 @@ func Create(label, workspaceID, description, clusterID, runtimeVersion string, s
 		if label == "" {
 			return errors.New("you must give your Deployment a name")
 		}
+		deployments, err := getDeployments(workspaceID, client)
+		if err != nil {
+			return errors.Wrap(err, errInvalidDeployment.Error())
+		}
+
+		for i := range deployments {
+			if deployments[i].Label == label {
+				return errors.New("A Deployment with that name already exists")
```

Comment:
trying to understand the workflow here, won't `client.createDeployment` call anyways throw this error?

Reply:
This function is not used when creating a deployment. client."logs/update/delete/etc..." takes a deployment id not a name so we need to make these checks ourselves to find the right deployment id